### PR TITLE
cset - allow NULL as value

### PR DESCRIPTION
**Pull request by wimleers**
PR for #5587.

### CODE REVIEW

File: src/Commands/config/ConfigCommands.php [hunk 1]
```diff
@@ -170,6 +171,11 @@ public function set($config_name, $key, $value, $options = ['input-format' => 's
             $data = [];
         }
 
+        // Special handling for empty array.
```
Comment (wimleers) [inline src/Commands/config/ConfigCommands.php]: 🙈 
```suggestion
        // Special handling for null.
```

File: src/Commands/config/ConfigCommands.php [hunk 2]
```diff
@@ -170,6 +171,11 @@ public function set($config_name, $key, $value, $options = ['input-format' => 's
...
+        if (strtolower($data) == 'null') {
+            $data = NULL;
```
Comment (wimleers) [inline src/Commands/config/ConfigCommands.php]: Apparently different coding style than Drupal! We need this:
```suggestion
            $data = null;
```

**Comment by weitzman** [comment]
Welcome to Drush, @wimleers!

**Comment by T2L** [comment]
this change produces

`Error: ]  TypeError: strtolower(): Argument #1 ($string) must be of type string, array given in strtolower() (line 175 of /vendor/drush/drush/src/Commands/config/ConfigCommands.php`

when using something like `./vendor/bin/drush config:set config_ignore.settings ignored_config_entities '[]' --yes` with PHP8.1+

**Pull request merged**